### PR TITLE
adjust area chart styling and aspect

### PR DESCRIPTION
**Pull request by hossamElfarFactorial**
## 🚪 Why?
- I needed to add a new aspect to the area chart.
- The current line type caused the area chart lines to goes above the max value or below the min value if the values between the ticks fluctuates between the min and max. 

## 🔑 What?
- Add a new aspect `small` and use by default for the area chart widget
- Add `recharts` line types and use `monotoneX` as a default line type
- Fix the summary value when the value is decimal
- Always show the x axis ticks by setting `interval` to 0. This is required as using the default value `preserveEnd ` could potential drop some ticks from the x axis.
- Show the area chart legend only if we have more than 1 area, and hide it otherwise
## 📸 Visual proof
image

### CODE REVIEW

File: lib/components/Charts/AreaChart/index.tsx [hunk 1]
```diff
@@ -25,7 +18,22 @@ export type AreaChartProps<
   DataConfig extends ChartConfig = ChartConfig,
   Keys extends string = InferChartKeys<DataConfig>,
 > = ChartPropsBase<DataConfig, Keys> & {
-  lineType?: "natural" | "linear" | "step"
+  lineType?:
```
Comment (elacallefactorial) [inline lib/components/Charts/AreaChart/index.tsx:21]: You might create a type in order no to have a long piece of code with the allowed line types

`type lineType =  "basis"  | "basisClosed" | "basisOpen" | "bumpX" | "bumpY" | "bump"....`

```
  DataConfig extends ChartConfig = ChartConfig,
  Keys extends string = InferChartKeys<DataConfig>,
> = ChartPropsBase<DataConfig, Keys> & {
  lineType?: lineType
```

File: lib/components/Widgets/Charts/ChartContainer.tsx [hunk 2]
```diff
@@ -20,18 +19,18 @@ const Container = forwardRef<
             <div className="mb-0.5 text-sm text-muted-foreground">
               {summary.label}
             </div>
-            <div className="flex flex-row items-end gap-0.5">
-              <ChartCounter value={summary.value} />
```
Comment (dani-moreno) [inline lib/components/Widgets/Charts/ChartContainer.tsx:24]: Why was this needed? if it is because of decimals, we can probably adjust the ChartCounter component.
Not sure about the unit being a prefix instead of a suffix. Maybe we need both cases?

File: lib/components/Charts/AreaChart/index.tsx [hunk 3]
```diff
@@ -21,11 +21,28 @@ import { fixedForwardRef } from "../utils/forwardRef"
 import { prepareData } from "../utils/muncher"
 import { ChartConfig, ChartPropsBase, InferChartKeys } from "../utils/types"
 
+type allowedLineTypes =
```
Comment (dani-moreno) [inline lib/components/Charts/AreaChart/index.tsx:24]: We probably don’t need all of them, right? Not entirely sure what the monotoneX one does, but it seems to get the job done. Maybe we can stick with that and simplify the options?
(This would be a great topic to discuss with Desi).

File: lib/components/Charts/AreaChart/index.tsx [hunk 4]
```diff
@@ -118,11 +136,13 @@ export const _AreaChart = <
             strokeWidth={1.5}
           />
         ))}
-        <ChartLegend
-          className={cn("flex justify-start", fullWidth && "ml-10")}
-          iconType="star"
-          content={<ChartLegendContent />}
-        />
+        {Object.keys(dataConfig).length > 1 && (
```
Comment (dani-moreno) [inline lib/components/Charts/AreaChart/index.tsx:139]: Yay, perfect! 🙌

**Review by dani-moreno — CHANGES_REQUESTED** [review]

File: lib/components/Charts/AreaChart/index.tsx (hunk 3 above)
Comment (hossamElfarFactorial) [reply lib/components/Charts/AreaChart/index.tsx]: [remove extra area chart line charts](https://github.com/factorialco/factorial-one/pull/449/commits/33c7ed9153eb2c955673b76044dcf202484c3d95)

File: lib/components/Widgets/Charts/ChartContainer.tsx (hunk 2 above)
Comment (hossamElfarFactorial) [reply lib/components/Widgets/Charts/ChartContainer.tsx:24]: [fix chart counter decimal points and add post and prefix units to cha…](https://github.com/factorialco/factorial-one/pull/449/commits/fd1c0f890c4f58327ef8c7c17ad55932c69d3f26)

**Review by dani-moreno — APPROVED** [review]
Nice, looks good @hossamElfarFactorial !

The ChartCounter animation looks a bit weird with decimals now, but that's something we can take a look in another PR!

**Comment by Rogermax** [comment]
I have a PR that makes charts reponsive (without aspect-ratio), can you check this PR: https://github.com/factorialco/factorial-one/pull/454 ?
